csp_report.php

```diff
@@ -159,8 +159,21 @@
     $where = '1 = 1';
     $params = array();
 }
-$table->set_sql($fields, $from, $where, $params);
 
+if (!$table->is_downloading()) {
+    echo $OUTPUT->header();
+    echo $OUTPUT->heading($title);
+
+    $action = new \confirm_action(get_string('areyousuretodeleteallrecords', 'local_csp'));
+    $urlresetallcspstatistics = new moodle_url($PAGE->url, array(
+        'resetallcspstatistics' => 1,
+        'sesskey' => sesskey(),
+    ));
+    echo $OUTPUT->single_button($urlresetallcspstatistics,
+        get_string('resetallcspstatistics', 'local_csp'), 'post', array('actions' => array($action)));
+}
+
+$table->set_sql($fields, $from, $where, $params);
 $table->out(30, true);
 
 echo $OUTPUT->footer();
```

Comment:
shouldn't the footer be conditional as well?

Reply:
No if you are exporting `$table->out` calls exit once it has gathered the output buffer and downloaded it